### PR TITLE
Feature/initial work

### PR DESCRIPTION
**Pull request by aussieveen**
Main features
- Auto-configuration of calls to Identity via env vars or overwritten in application services.yaml file
- AccessTokenCache which will cache a service token based on the scopes it has. 
- Support for Access and Account apis

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!groovy
```
Comment (Rayrn) [inline Jenkinsfile:1]: Why is this here? Jenkins can't access public github. We should be using GH Actions. Also, this should probably be under github.com/immediatemediaco.

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "php-http/client-common": "^2.4",
+        "php-http/discovery": "^1.19.0",
+        "psr/http-client": "^1.0",
+        "psr/http-factory": "^1.0",
+        "symfony/cache": "^6.0",
+        "symfony/dependency-injection": "^6.0",
+        "symfony/http-kernel": "^6.0"
+    },
+    "require-dev": {
+        "10up/wp_mock": "0.4.2",
```
Comment (Rayrn) [inline composer.json:18]: 1.0 plx

File: README.md [hunk 3]
```diff
@@ -1,2 +1,118 @@
...
+
+```php
+IM\Fabric\Bundle\IdentityApiBundle\IdentityApiBundle::class => ['all' => true]
+```
+
+If not, run `composer recipes:install immediate/im-identity-api-bundle --force`
+
+Alongside the package installation, you will also need to require packages that include implementations for PSR 7, PSR 17 and PSR 18 interfaces.
+
+## Configuration
```
Comment (Rayrn) [inline README.md:34]: Can we update this to use `Configuration: Symfony` & `Configuration: WordPress` sections please (assuming from the wp_mock dependency it supports both.

File: src/Api/AbstractApi.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        return $this->parseResponse($this->client->prepareHttpClient()->post(
+            $path,
+            $requestHeaders,
+            $body
+        ));
+    }
+
+    private function parseResponse(ResponseInterface $response): array
+    {
+        return json_decode($response->getBody()->getContents(), true);
```
Comment (Rayrn) [inline src/Api/AbstractApi.php:44]: Its probably worth noting somewhere that we're expecting the downstream API to return json.

File: Jenkinsfile (hunk 1 above)
Comment (aussieveen) [reply Jenkinsfile:1]: :doh: I'lll get some actions in. I put it here to avoid the pain that comes with the other repo which I understand isn't supported yet.

File: composer.json (hunk 2 above)
Comment (aussieveen) [reply composer.json:18]: Technically not needed at all so I'll just remove outright.

File: README.md (hunk 3 above)
Comment (aussieveen) [reply README.md:34]: It does not. I'm removing the `wp_mock` for now.

File: src/Api/AbstractApi.php (hunk 4 above)
Comment (aussieveen) [reply src/Api/AbstractApi.php:44]: Fair. I'll add a comment.

File: Jenkinsfile (hunk 1 above)
Comment (Rayrn) [reply Jenkinsfile:1]: Might be worth adding in here:
https://github.com/immediate-media/github-actions/tree/main/.github/workflows

**Review by Rayrn — APPROVED** [review]
